lib/android/ui/ui.nit

```diff
@@ -42,28 +42,64 @@ redef class NativeActivity
 	private fun remove_title_bar in "Java" `{
 		self.requestWindowFeature(android.view.Window.FEATURE_NO_TITLE);
 	`}
+
+	# Insert a single layout as the root of the activity window
+	private fun insert_root_layout(root_layout_id: Int)
+	in "Java" `{
+		android.widget.FrameLayout layout = new android.widget.FrameLayout(self);
+		layout.setId((int)root_layout_id);
+		self.setContentView(layout);
+	`}
+
+	# Replace the currently visible fragment, if any, with `native_fragment`
+	private fun show_fragment(root_layout_id: Int, native_fragment: Android_app_Fragment)
+	in "Java" `{
+		android.app.FragmentTransaction transaction = self.getFragmentManager().beginTransaction();
+		transaction.replace((int)root_layout_id, native_fragment);
+		transaction.commit();
+	`}
 end
 
 redef class App
 	redef fun on_create
 	do
 		app.native_activity.remove_title_bar
+		native_activity.insert_root_layout(root_layout_id)
+		super
+	end
+
+	# Identifier of the container holding the fragments
+	private var root_layout_id = 0xFFFF
+
+	redef fun window=(window)
+	do
+		native_activity.show_fragment(root_layout_id, window.native)
 		super
 	end
 end
 
+# On Android, a window is implemented with the fragment `native`
 redef class Window
-	redef var native = app.native_activity.new_global_ref
+	redef var native = (new Android_app_Fragment(self)).new_global_ref
 
-	redef type NATIVE: NativeActivity
+	redef type NATIVE: Android_app_Fragment
+
+	# Root high-level view of this window
+	var view: nullable View = null
 
 	redef fun add(item)
 	do
+		if item isa View then view = item
```

Comment:
I had the same doubt. But, in theory, the window could have other non-view controls (like a key catcher) which would still use the `add` control. So like this, the window can use the same services of other composite controls.